src/clr-angular/utils/drag-and-drop/draggable.ts

```diff
@@ -3,40 +3,82 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {AfterContentInit, Directive, ElementRef, OnDestroy} from "@angular/core";
+import {
+    AfterContentInit,
+    ComponentFactory,
+    ComponentFactoryResolver,
+    ContentChild,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Injector,
+    OnDestroy,
+    Output,
+    ViewContainerRef
+} from "@angular/core";
 import {Subscription} from "rxjs/Subscription";
+
+import {DomAdapter} from "../dom-adapter/dom-adapter";
+import {ClrDraggableGhost} from "./draggable-ghost";
+import {ClrIfDragged} from "./if-dragged";
 import {ClrDragEvent} from "./interfaces/drag-event";
 import {ClrDragEventListener} from "./providers/drag-event-listener";
 import {ClrDragHandleRegistrar} from "./providers/drag-handle-registrar";
+import {ClrDraggableStateRegistrar} from "./providers/draggable-state-registrar";
 
-
-@Directive(
-    {selector: "[clrDraggable]", providers: [ClrDragEventListener, ClrDragHandleRegistrar], host: {class: "draggable"}})
+@Directive({
+    selector: "[clrDraggable]",
+    providers: [ClrDragEventListener, ClrDragHandleRegistrar, ClrDraggableStateRegistrar, DomAdapter],
+    host: {class: "draggable"}
+})
 export class ClrDraggable<T> implements AfterContentInit, OnDestroy {
     private draggableEl: Node;
-
     private subscriptions: Subscription[] = [];
+    private componentFactory: ComponentFactory<ClrDraggableGhost<T>>;
 
     constructor(private el: ElementRef, private dragEventListener: ClrDragEventListener<T>,
-                private dragHandleRegistrar: ClrDragHandleRegistrar<T>) {
+                private dragHandleRegistrar: ClrDragHandleRegistrar<T>, private viewContainerRef: ViewContainerRef,
+                private cfr: ComponentFactoryResolver, private injector: Injector,
+                private draggableStateRegistrar: ClrDraggableStateRegistrar<T>) {
         this.draggableEl = this.el.nativeElement;
+        this.componentFactory = this.cfr.resolveComponentFactory<ClrDraggableGhost<T>>(ClrDraggableGhost);
     }
 
+    @ContentChild(ClrIfDragged) customGhost: ClrDraggableGhost<T>;
```

Comment:
You're querying `ClrIfDragged` but you're typing what you receive as `ClrDraggableGhost`. I have no idea why this doesn't throw an error in AoT, but it should. What you want is:
```typescript
@ContentChild(ClrIfDragged) customGhost: ClrIfDragged<T>;
```

Reply:
Ooh right, good catch! 

On the side, I am also wondering it could be the way it is in Angular? Because `clrIfDragged` is placed on the `clr-draggable-ghost` component. So it's kind of both `IfDraggad` and `ClrDraggableGhost` at the same time. Nevertheless, I will change according to your suggestion.

Reply:
Addressed this issue.